### DISCUSSION TITLE
Where flusing database after successful user login to retain login history ?

### DISCUSSION THREAD

**Discussion by sdespont** [symfony/symfony#47908] [Q&A]
I need to save the login date and time of the website's users after their login.

Because I am using `form_login` authentication manager, where is the best place to save this data in the database ? 

Is it OK to do that in the `success_handler` by injecting EntityManagerInterface and flush `User` entity in `onAuthenticationSuccess`  ?

I am asking because I know it is not a good practice to flush database in a listener.

**Comment by sdespont** [comment]
Any advice?

**Comment by ker0x** [comment]
You can listen to the `LoginSuccessEvent`

https://symfony.com/doc/current/security.html#authentication-events
